components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -877,6 +878,166 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
         }
     }
 
+    /**
+     * Update the SCIM user.
+     *
+     * @param user                           {@link User} object.
+     * @param requiredAttributes             A map of required attributes in SCIM schema.
+     * @param allSimpleMultiValuedAttributes A List of simple multi-valued attributes in SCIM schema.
+     * @return The updated user.
+     * @throws CharonException     Exception occurred in charon level.
+     * @throws BadRequestException Exception occurred due to a bad request.
+     */
+    public User updateUser(User user, Map<String, Boolean> requiredAttributes,
+                           List<String> allSimpleMultiValuedAttributes) throws CharonException, BadRequestException {
+
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Updating user: " + user.getUserName());
+            }
+
+             /* Set thread local property to signal the downstream SCIMUserOperationListener
+                about the provisioning route. */
+            SCIMCommonUtils.setThreadLocalIsManagedThroughSCIMEP(true);
+            // Get user claim values.
+            Map<String, String> claims = AttributeMapper.getClaimsMap(user);
+            // Get claims mapped with simple multi-valued attributes.
+            Map<String, String> allSimpleMultiValuedClaims = new HashMap<>();
+
+            for (String simpleMultiValuedAttribute : allSimpleMultiValuedAttributes) {
+                allSimpleMultiValuedClaims.put(simpleMultiValuedAttribute, "");
```

Comment:
Shall we use String.EMPTY here?

Reply:
addressed in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/266/commits/d419b5d649f8bc9093b6c4143d0fa188cce5ad99